pkg/storage/pvc.go

```diff
@@ -20,9 +28,136 @@ type PVCBuilder struct {
 
+	errorMsg  string
 	apiClient *clients.Settings
 }
 
+// NewPVCBuilder creates a new structure for persistentvolumeclaim.
+func NewPVCBuilder(apiClient *clients.Settings, name, nsname string) *PVCBuilder {
+	glog.V(100).Infof("Creating PersistentVolumeClaim %s in namespace %s",
+		name, nsname)
+	builder := PVCBuilder{
+		Definition: &v1.PersistentVolumeClaim{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{},
+		},
+	}
+
+	builder.apiClient = apiClient
+
+	if name == "" {
+		glog.V(100).Infof("PVC name is emtpty")
+		builder.errorMsg = "PVC name is empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("PVC namespace is emtpty")
+		builder.errorMsg = "PVC namespace is empty"
+	}
+
+	return &builder
+
+}
+
+// WithPVCAccessMode configure access mode for the PV.
+func (builder *PVCBuilder) WithPVCAccessMode(accessMode string) (*PVCBuilder, error) {
+	glog.V(100).Infof("Set PVC accessMode: %s", accessMode)
+
+	if accessMode == "" {
+		glog.V(100).Infof("Empty accessMode for PVC %s", builder.Definition.Name)
+		builder.errorMsg = "Empty accessMode for PVC requested"
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	if !validatePVCAccessMode(accessMode) {
+		glog.V(100).Infof("Invalid accessMode for PVC %s", accessMode)
+		builder.errorMsg = fmt.Sprintf("Invalid accessMode for PVC %s", accessMode)
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	if builder.Definition.Spec.AccessModes != nil {
+		builder.Definition.Spec.AccessModes = append(builder.Definition.Spec.AccessModes,
+			v1.PersistentVolumeAccessMode(accessMode))
+	} else {
+		builder.Definition.Spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.PersistentVolumeAccessMode(accessMode)}
+	}
+
+	return builder, nil
+
+}
+
+// validatePVCAccessMode validates if requested mode is valid for PVC.
+func validatePVCAccessMode(accessMode string) bool {
+	glog.V(100).Info(fmt.Sprintf("Validating accessMode %s", accessMode))
+
+	_, ok := validPVCModesMap[accessMode]
+	return ok
+
+}
+
+func (builder *PVCBuilder) WithPVCapacity(capacity string) (*PVCBuilder, error) {
+	if capacity == "" {
+		glog.V(100).Infof("Capacity of the PersistentVolumeClaim is empty")
+		builder.errorMsg = "Capacity of the PersistentVolumeClaim is empty"
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	defer func() (*PVCBuilder, error) {
+		if r := recover(); r != nil {
+			glog.V(100).Infof("Failed to parse %v", capacity)
+			builder.errorMsg = fmt.Sprintf("Failed to parse: %v", capacity)
+			return builder, fmt.Errorf(fmt.Sprintf("Failed to parse: %v", capacity))
+		}
+		return builder, nil
+	}()
+
+	capMap := make(map[v1.ResourceName]resource.Quantity)
+	capMap[v1.ResourceStorage] = resource.MustParse(capacity)
+
+	builder.Definition.Spec.Resources = v1.ResourceRequirements{Requests: capMap}
+
+	return builder, nil
+
+}
+
+func (builder *PVCBuilder) WithStorageClass(storageClass string) (*PVCBuilder, error) {
```

Comment:
Missing comment. The linter on this project does not check that.

Reply:
thanks, updated